src/main/java/ch/njol/skript/expressions/ExprEntityOwner.java

```diff
@@ -0,0 +1,131 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Item;
+import org.bukkit.entity.Tameable;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+@Name("Entity Owner")
+@Description({
+	"The owner of a tameable entity (i.e. horse or wolf) or a dropped item.",
+	"NOTE: If the owner of a dropped item was an entity and was killed, will return their uuid in string form."
+})
+@Examples({
+	"set owner of target entity to player",
+	"delete owner of target entity",
+	"set {_t} to uuid of tamer of target entity",
+	"",
+	"set the owner of all dropped items to player"
+})
+@Since("2.5, INSERT VERSION (dropped items)")
+public class ExprEntityOwner extends SimplePropertyExpression<Entity, Object> {
+	
+	static {
+		Skript.registerExpression(ExprEntityOwner.class, Object.class, ExpressionType.PROPERTY,
+			"[the] (owner|tamer) of %livingentities%",
+			"[the] [[dropped] item] owner of %itementities%");
+	}
+
+	private boolean useTameable;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		useTameable = matchedPattern == 0;
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	public @Nullable Object convert(Entity entity) {
+		if (entity instanceof Tameable tameable && tameable.isTamed()) {
+			return tameable.getOwner();
+		} else if (entity instanceof Item item) {
+			UUID uuid = item.getOwner();
+			if (uuid == null)
+				return null;
+			Entity checkEntity = Bukkit.getEntity(uuid);
+			if (checkEntity != null)
+				return checkEntity;
+			OfflinePlayer checkPlayer = Bukkit.getOfflinePlayer(uuid);
+			if (checkPlayer != null)
+				return checkPlayer;
+			return uuid.toString();
```

Comment:
I worry about this.
Lets pretend Mr Skeleton dropped an item, and then he died or ran away.
So now we don't have an entity and we get to this line.
But we're checking for an offline player. Will this hang the server?

Additionally, as per the docs for Bukkit.getOfflinePlayer:
"This will return an object even if the player does not exist. To this method, all players will exist."
So were forcing an offline player object?!?!

I guess this is the battle Skript faces when Bukkit/Minecraft returns a UUID and we're trying to check if its something else.

Reply:
I'm open to any suggestions, I cant really think of anyway to properly  combat it

Reply:
Not a lot agree with me, but I personally feel Skript should do its best to match Bukkit/Minecraft when it can.
So with that said, in my **opinion** I personally think this should return UUIDs (as strings) rather than forcing it to be an entity/offline player.

Might want the rest of the team to weigh in on this.

Reply:
I would be okay with only returning loaded entities and online players, and none otherwise.
Returning only uuid strings would be okay too but it would mean changing the syntax to clarify that

Reply:
if no offlineplayer is found, it should definitely return null instead of the string uuid, since every other return is an entity. why should it suddenly return a uuid?

as for returning uuids, i think when using this expression, i'd expect to get the actual owning entity. getting the entity from the uuid is not something the user should worry about, only the owner being null

Reply:
> I would be okay with only returning loaded entities and online players, and none otherwise. Returning only uuid strings would be okay too but it would mean changing the syntax to clarify that

I dont know if you saw my other message or not, but it appears `Bukkit.getOfflinePlayer(UUID)` returns an offline player object regardless if there is one with that uuid.
So for instance, if the Owner (let's pretend a skeleton) is dead/not loaded, it's going to return an OP object, which doesn't properly reflect what Minecraft is actually doing here.

Reply:
> I dont know if you saw my other message or not, but it appears Bukkit.getOfflinePlayer(UUID) returns an offline player object regardless if there is one with that uuid.

Yeah, that's why I specified online players only

Reply:
well.. it appears I misread your message, my bad!

Reply:
> > I dont know if you saw my other message or not, but it appears Bukkit.getOfflinePlayer(UUID) returns an offline player object regardless if there is one with that uuid.
> 
> 
> 
> Yeah, that's why I specified online players only

Perhaps cached offline players? Iirc there is a method for it